truth_set/python/structure/solution-approach/standard-method/OrderProcessor.py

```diff
@@ -0,0 +1,19 @@
+class OrderProcessor:
+    def apply_discounts(self, order_total, discounts_applied):
+        if discounts_applied > 5:
+            raise Exception("Too many discounts applied.")
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Create custom exception class</b></div>
<div id="fix">

Consider creating a custom exception class instead of using generic 'Exception'.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -1,4 +1,8 @@
 +class TooManyDiscountsError(Exception):
 +    """Raised when too many discounts are applied to an order."""
 +    pass
 +
  class OrderProcessor:
 -        raise Exception("Too many discounts applied.")
 +        raise TooManyDiscountsError("Maximum discount limit exceeded")
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/5#issuecomment-2553425774>#746476</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged